### PR TITLE
Add API server config.download functionality.

### LINKED ISSUE #1514: [FR]: API server - provide endpoint "config download" to download original configuration
The current API call config.list will output the configuration as JSON by design. Provide endpoint that will output original Turbinia config file so it can be edited/re-used with Turbinia.

### PR DESCRIPTION
**Pull request by hacktobeer**
### Description of the change
This PR adds a config.download API end point to download the Turbinia configuration file in it's original format.

### Applicable issues

- fixes #1514 

### Checklist

- [x] All tests were successful.
- [x] Unit tests added.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by hacktobeer** [comment]
@jleaniz first API server contribution. Can you have a look and see what needs to be added/changed? 
Probably:
- unit tests
- do I need to generate the client? if so, how?

File: turbinia/api/routes/config.py [hunk 1]
```diff
@@ -58,3 +59,16 @@ async def get_version(request: Request):
...
   response = {'version': __version__}
   return JSONResponse(content=response, status_code=200)
+
+
+@router.get('/download')
+async def download_config(request: Request):
+  """Downloads Turbinia configuration."""
+  try:
+    content = api_utils.get_configuration_content()
+    return Response(content=content, status_code=200)
```
Comment (jleaniz) [inline turbinia/api/routes/config.py:69]: This should work. However, FastAPI provides a ```FileResponse``` type you can use. It should simplify the code quite a bit. https://fastapi.tiangolo.com/advanced/custom-response/#fileresponse

Comment (jleaniz) [reply turbinia/api/routes/config.py:69]: Here is an example for a unit test. The client is already prepared and you just have to call self.client.get(...) and compare the response to what's expected.

https://github.com/google/turbinia/blob/master/turbinia/api/api_server_test.py#L206

**Review by jleaniz — CHANGES_REQUESTED** [review]
Thanks for the PR! Here's some comments for you to review.

**Comment by hacktobeer** [comment]
@jleaniz PTAL -> tests added and rewritten to use FileResponse fastapi function.

File: turbinia/api/routes/config.py [hunk 2]
```diff
@@ -18,11 +18,12 @@
 import logging
 
 from fastapi import HTTPException, APIRouter
-from fastapi.responses import JSONResponse
+from fastapi.responses import JSONResponse, FileResponse
 from fastapi.requests import Request
 
 from turbinia import __version__
 from turbinia import config as turbinia_config
+from turbinia.api import utils as api_utils
```
Comment (jleaniz) [inline turbinia/api/routes/config.py:26]: I don't think this import is needed anymore.

**Review by jleaniz — APPROVED** [review]
LGTM, just a minor comment. Congrats on your first API server PR :)

**Pull request merged**